tapdb/assets_store.go

```diff
@@ -3526,6 +3548,90 @@ func (a *AssetStore) TxHeight(ctx context.Context, txid chainhash.Hash) (uint32,
 	return uint32(dbBlockHeight), nil
 }
 
+// InsertBurn inserts a burnt asset to the db which includes all related
+// metadata.
+func (a *AssetStore) InsertBurn(ctx context.Context, txid []byte, note string,
+	assetID, groupKey []byte, amount uint64) error {
+
+	var writeTxOpts AssetStoreTxOptions
+	return a.db.ExecTx(ctx, &writeTxOpts, func(q ActiveAssetsStore) error {
+		// Look up the transfer that relates to this txid.
+		// TODO(george): We assume there's a single burn in every txn,
```

Comment:
I don't think we need this TODO. If there are multiple burns in a transaction, we'd call `InsertBurn` multiple times with the same `txid`. But because they would all still be in the same transfer, we'd query the same transfer ID which is correct.

Reply:
^ makes sense

Reply:
> But because they would all still be in the same transfer, we'd query the same transfer ID which is correct.

Ah right, I believed that there would be multiple transfers which is not the way it work